elasticsearch/src/http/transport.rs

```diff
@@ -474,6 +519,102 @@ impl Transport {
         if let Some(q) = query_string {
             request_builder = request_builder.query(q);
         }
+        Ok(request_builder)
+    }
+
+    fn parse_to_url(address: &str, scheme: &str) -> Result<Url, Error> {
+        if address.is_empty() {
+            return Err(crate::error::lib("Bound Address is empty"));
+        }
+
+        let matches = ADDRESS_REGEX
```

Comment:
I am a little uncomfortable using regex based parsing here. Wouldn't it be easier to check the `address` str for a valid scheme prefix? That would be `http://` `https://` and if it doesn't have one, prepend the given `scheme`, parse the url and then use `Url::set_scheme`?